File: cpp/components/validation-ctt/HEAD/ctt_testcases/IntrospectionAnnotation.cpp
/******************************************************************************
* * 
*    Copyright (c) 2016 Open Connectivity Foundation and AllJoyn Open
*    Source Project Contributors and others.
*    
*    All rights reserved. This program and the accompanying materials are
*    made available under the terms of the Apache License, Version 2.0
*    which accompanies this distribution, and is available at
*    http://www.apache.org/licenses/LICENSE-2.0

******************************************************************************/
#include "stdafx.h"
#include "IntrospectionAnnotation.h"

using namespace std;

string IntrospectionAnnotation::getValue() const
{
	return m_Value;
}

void IntrospectionAnnotation::setValue(string t_Value)
{
	m_Value = t_Value;
}

int IntrospectionAnnotation::compareTo(IntrospectionAnnotation t_IntrospectionAnnotation)
{
	return getName() == t_IntrospectionAnnotation.getName();
}

bool IntrospectionAnnotation::operator <(const IntrospectionAnnotation& t_IntrospectionAnnotation) const
{
	if (getName().compare(t_IntrospectionAnnotation.getName()) < 0)
	{
		return true;
	}
	else if (getName().compare(t_IntrospectionAnnotation.getName()) > 0)
	{
		return false;
	}
	else
	{
		return (getValue().compare(t_IntrospectionAnnotation.getValue()) < 0);
	}
}

IntrospectionAnnotation::operator std::string() const
{
	string stringToReturn;
	stringToReturn.append("Annotation [name=");
	stringToReturn.append(getName());
	stringToReturn.append(", Value=");
	stringToReturn.append(getValue());
	stringToReturn.append("]");

	return stringToReturn;
}

File: java/components/validation-ctt/HEAD/ctt_testcases/src/com/at4wireless/alljoyn/core/introspection/bean/IntrospectionAnnotation.java
/*******************************************************************************
 *   *  
 *     Copyright (c) 2016 Open Connectivity Foundation and AllJoyn Open
 *     Source Project Contributors and others.
 *     
 *     All rights reserved. This program and the accompanying materials are
 *     made available under the terms of the Apache License, Version 2.0
 *     which accompanies this distribution, and is available at
 *     http://www.apache.org/licenses/LICENSE-2.0

 *******************************************************************************/
package com.at4wireless.alljoyn.core.introspection.bean;

public class IntrospectionAnnotation extends IntrospectionBaseTag implements Comparable<IntrospectionAnnotation>
{
    private String value;

    public String getValue()
    {
        return value;
    }

    public void setValue(String value)
    {
        this.value = value;
    }

    @Override
    public int compareTo(IntrospectionAnnotation introspectionAnnotation)
    {
        return getName().compareTo(introspectionAnnotation.getName());
    }

    @Override
    public int hashCode()
    {
        final int prime = 31;
        int result = super.hashCode();
        result = prime * result + ((value == null) ? 0 : value.hashCode());

        return result;
    }

    @Override
    public boolean equals(Object obj)
    {
        if (this == obj)
        {
            return true;
        }
        if (!super.equals(obj))
        {
            return false;
        }
        if (!(obj instanceof IntrospectionAnnotation))
        {
            return false;
        }
        
        IntrospectionAnnotation other = (IntrospectionAnnotation) obj;
       
        if (value == null)
        {
            if (other.value != null)
            {
                return false;
            }
        }
        else if (!value.equals(other.value))
        {
            return false;
        }

        return true;
    }

    @Override
    public String toString()
    {
        StringBuilder builder = new StringBuilder();
        builder.append("Annotation [name=");
        builder.append(getName());
        builder.append(", value=");
        builder.append(value);
        builder.append("]");

        return builder.toString();
    }
}